### DISCUSSION TITLE
Run Templater outside Obsidian

### DISCUSSION THREAD

**Discussion by torViadee** [SilentVoid13/Templater#1189] [Ideas & Feature Requests]
In my Workflow, I need to create some Files in my Vault with Scripts. I would like to use Templater in these Sctipts to generate those Files when Obsidian is not running.

My Questions:
a) is it alrady possible to run Templater outside of Obsidian? Are the excamples available?
b) if not, is this a planned feature?

**Comment by Zachatoo** [comment]
No Obsidian plugin can run when Obsidian isn't running. It's a limitation of the Obsidian plugin ecosystem. Nothing to be done about it.

**Reply by Zachatoo** [reply]
It is possible to use the Obsidian URI protocol to send messages to Obsidian to create files from outside of Obsidian. I believe the Advanced URI plugin supports running commands, including Templater commands.

**Reply by AB1908** [reply]
This being said, you could redesign your scripts to be user scripts and then call those JS files outside Obsidian with whatever. This is heavily dependent on the types of templates you have though.

**Reply by Zachatoo** [reply]
True, if your user scripts don't rely on any APIs from Templater, Obsidian, or other plugins, you're golden. You could use more standard filesystem APIs from your JS framework of choice instead.

**Reply by brandonmcclure** [reply]
> It is possible to use the Obsidian URI protocol to send messages to Obsidian to create files from outside of Obsidian. I believe the Advanced URI plugin supports running commands, including Templater commands.

Thanks for the suggestion. I played around with this plugin today and wanted to share what I learned. I was able to create a note and launch the templater New note modal with the URI below. I could not figure out how to pass the template name in via the URI, so the process is manual still and not sufficient for running non interactively in a script. I tried passing the parameters for the [create_new_note_from_template](https://github.com/SilentVoid13/Templater/blob/d07091472560f0a76dfb7e5f239916504939c350/src/core/Templater.ts#L110C11-L110C40) class as URI parameters, but that did not seem to work. 

```
obsidian://advanced-uri?vault=myVault&filepath=path%252Fwith%2520spaces%252F2FmyNote.md&commandid=templater-obsidian%253Acreate-new-note-from-template
```

There is nothing in the advanced uri [documentation](https://vinzent03.github.io/obsidian-advanced-uri/actions/commands) for passing arguments to commands, and [this issue](https://github.com/Vinzent03/obsidian-advanced-uri/issues/126) would seem to indicate that it is not possible from that plugin. 

I did find that if I have `Folder Templates` enabled, then I can just use the base uri to create a new file, and templater will create the file without interaction. This is sufficient for my needs, but I could see times I would want to have a external process create a new note from templater in a location that has not been predefined

```
obsidian://new?vault=myVault&file=path%2Fwith%20spaces%2FmyNote.md
```
Notice that the uri params and encoding is different between advanced-uri and the base. I used the `copy Obsidian/Advanced URI` button quite a bit to find the URIs for my specific files.
